### PR TITLE
Correct the usage of bool and str

### PR DESCRIPTION
**Pull request by ganhuang**
In ansible, `'str' | bool` is False:

```
$ cat test.yml 
---
- hosts: localhost
  gather_facts: false
  tasks:
   - set_fact:
       test_net_cidr: "172.30.0.0.1/16"

   - debug:
       msg: "String test_net_cidr is bool False"
     when: not test_net_cidr | bool


$ ansible-playbook test.yml 
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [localhost] ************************************************************************************************************************************************************

TASK [set_fact] *************************************************************************************************************************************************************
ok: [localhost]

TASK [debug] ****************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false
}

MSG:

String test_net_cidr is bool False


PLAY RECAP ******************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0   

```

For a string variable (default to `False`), we should just use `{% if some_string %}`

For a bool value, we use something like `{% if any_bool | bool %}`

### CODE REVIEW

File: roles/container_runtime/tasks/package_docker.yml [hunk 1]
```diff
@@ -111,12 +111,12 @@
     regexp: '^OPTIONS=.*$'
     line: "OPTIONS='\
       {% if ansible_selinux.status | default(None) == 'enabled' and openshift_docker_selinux_enabled | default(true) | bool %} --selinux-enabled {% endif %} \
-      {% if openshift_docker_log_driver | bool %} --log-driver {{ openshift_docker_log_driver }}{% endif %} \
+      {% if openshift_docker_log_driver %} --log-driver {{ openshift_docker_log_driver }}{% endif %} \
```
Comment (sdodson) [inline roles/container_runtime/tasks/package_docker.yml]: I imagine the most clear way write this is `if openshift_docker_log_driver is defined` ?

Comment (ganhuang) [reply roles/container_runtime/tasks/package_docker.yml]: @sdodson `openshift_docker_log_driver` is already defined and defaults to `False`

https://github.com/openshift/openshift-ansible/blob/master/roles/container_runtime/defaults/main.yml#L28

**Comment by ganhuang** [comment]
@michaelgugino Appended a commit that `openshift_docker_hosted_registry_network_default` defaults to `openshift.common.portal_net` while openshift_portal_net is not set in inventory file.

PTAL, thanks!

**Comment by ganhuang** [comment]
> openshift_portal_net became mandatory in 3.7.

I can't see any code doing it yet, could you kindly point it out?

**Comment by michaelgugino** [comment]
@ganhuang We are indeed still using openshift.common.portal_net in places; however, it will be removed, and openshift_portal_net is required as documented: https://github.com/openshift/openshift-ansible/blob/master/inventory/hosts.example#L748

We made openshift_portal_net required because it's too error-prone trying to guess what an existing cluster's portal_net might be, so we are hard-requiring that openshift_portal_net is supplied by the user.

**Comment by ganhuang** [comment]
Closing in favor of https://github.com/openshift/openshift-ansible/pull/7184

**Pull request closed without merging**